#DOAJ Publisher Frequencies
This document is a report showing how many journal has each **publishers** The CSV file were taken from the DOAJ website. One in 2014 and the other one in 2015. We will compare both file. To view the blog file that comes with this document please refer to our blog [SustainingKnowledgeCommons.org](http://sustainingknowledgecommons.org/)
|
+ Supervisor: **Heather Morrison**
+ Author: **Guinlsy Mondésir**
+ Date: 2015-09-22
+ Project: **[SustainingKnowledgeCommons](http://sustainingknowledgecommons.org/)**


<hr/>

##Frequency for 2015

In [2]:
import pandas as pd
from pandas import DataFrame
from pandas import Series

In [3]:
doaj_2015 = pd.read_csv('doaj_20151015_1800_utf8.csv', encoding='utf-8', dtype='unicode')
publishers = doaj_2015['Publisher'].value_counts()

In [33]:
publishers = DataFrame(publishers.reset_index())
publishers.columns = ['Publisher','Frequency_2015']

In [34]:
publishers.head(20)

Publisher  Frequency_2015
0                    Hindawi Publishing Corporation             539
1                                    BioMed Central             271
2                                   De Gruyter Open             212
3                                           MDPI AG             121
4                                          Springer             108
5                                Dove Medical Press             105
6                                      Bentham open              81
7                              Medknow Publications              78
8                                          Elsevier              72
9                                Libertas Academica              56
10                                  Frontiers Media              49
11            Internet Scientific Publications, LLC              46
12                          Copernicus Publications              43
13                           PAGEPress Publications              42
14                 Universidad Nacional de Colombia              35
15                        Universidade de São Paulo              33
16                                            Wiley              33
17  Consejo Superior de Investigaciones Científicas              31
18                             Co-Action Publishing              31
19                                SAGE Publications              28

In [35]:
publishers.to_excel('publisher.xlsx', 
                 sheet_name = 'publisher frequencies', 
                 index = False)

###Verification in Ruby
the following cell are the equivalent in Ruby (2.1.0) for the 2015 doaj_metadata file. The different numbers in some of the publisher frequency (i.e. MDPI) is because that 'smarter_csv' does trim whitespace and **pandas** not by default.

    #gem install smarter_csv
    #gem install pry
    require 'smarter_csv'
    require 'pry'


    doaj_file_2015 = SmarterCSV.process('doaj_20151015_1800_utf8.csv')
    publishers = []
    doaj_file_2015.each { |data| publishers << data[:publisher] }
    results = publishers.inject(Hash.new(0)) {|hash, arr_element| hash[arr_element] += 1; hash }
    results = results.sort_by {|_key, value| value}.to_h


    last_twenty =  Hash[results.sort_by { |k,v| -v }[0..20]]
    Pry::ColorPrinter.pp(last_twenty)

    ####Results
    =begin 

        {"Hindawi Publishing Corporation"=>539,
         "BioMed Central"=>278,
         "De Gruyter Open"=>212,
         "MDPI AG"=>127,
         "Springer"=>108,
         "Dove Medical Press"=>105,
         "Medknow Publications"=>85,
         "Bentham open"=>81,
         "Elsevier"=>72,
         "Libertas Academica"=>56,
         "Frontiers Media"=>49,
         "Internet Scientific Publications, LLC"=>46,
         "Copernicus Publications"=>43,
         "PAGEPress Publications"=>42,
         "Universidade de São Paulo"=>37,
         "Universidad Nacional de Colombia"=>35,
         "Wiley"=>33,
         "Co-Action Publishing"=>31,
         "Consejo Superior de Investigaciones Científicas"=>31,
         "Tehran University of Medical Sciences"=>31,
         "SAGE Publications"=>28}

        =end

<hr/>

##Frequency for 2014

In [36]:
doaj_2015 = pd.read_csv('doaj_20140507_1330_utf8.csv', encoding = "ISO-8859-1")
publishers_2014 = doaj_2015['Publisher'].value_counts()

###notes
We had some issues with the 2014 csv file with unicodes we couldn't find a proper way to decode all the caracter with the proper encoding value. So that is the reason why some Publisher name have typo: Universidade de **Sí£o Paulo**

In [37]:
publishers_2014 = DataFrame(publishers_2014.reset_index())
publishers_2014.columns = ['Publisher','Frequency_2014']

In [38]:
publishers_2014.head(20)

Publisher  Frequency_2014
0             Hindawi Publishing Corporation             438
1                             BioMed Central             238
2             Scientific Research Publishing             119
3                               Bentham open              99
4                                    MDPI AG              95
5                                   Springer              95
6                         Dove Medical Press              91
7                       Medknow Publications              80
8                         Libertas Academica              49
9                     PAGEPress Publications              47
10                                  Versita               46
11     Internet Scientific Publications, LLC              46
12                   Copernicus Publications              35
13          Universidad Nacional de Colombia              34
14  Canadian Center of Science and Education              34
15                                   Versita              31
16                      Co-Action Publishing              31
17                Universidade de Sí£o Paulo              31
18                           Frontiers Media              30
19     Tehran University of Medical Sciences              30

In [39]:
publishers_2014.to_excel('publisher_2014.xlsx', 
                 sheet_name = 'publisher frequencies', 
                 index = False)

##Ruby
I couldn't do the frequency for the 2014 because it has unicode errors

##Concatenation

In [40]:
publishers_2015 = publishers
df1 = DataFrame(publishers_2015.head(30))
df2 = DataFrame(publishers_2014.head(30))
frames = [df1, df2]
result = pd.concat(frames,axis=1)
result

Publisher  Frequency_2015  \
0                      Hindawi Publishing Corporation             539   
1                                      BioMed Central             271   
2                                     De Gruyter Open             212   
3                                             MDPI AG             121   
4                                            Springer             108   
5                                  Dove Medical Press             105   
6                                        Bentham open              81   
7                                Medknow Publications              78   
8                                            Elsevier              72   
9                                  Libertas Academica              56   
10                                    Frontiers Media              49   
11              Internet Scientific Publications, LLC              46   
12                            Copernicus Publications              43   
13                             PAGEPress Publications              42   
14                   Universidad Nacional de Colombia              35   
15                          Universidade de São Paulo              33   
16                                              Wiley              33   
17    Consejo Superior de Investigaciones Científicas              31   
18                               Co-Action Publishing              31   
19                                  SAGE Publications              28   
20              Tehran University of Medical Sciences              28   
21                                           GERFLINT              24   
22  Academy & Industry Research Collaboration Cent...              23   
23                              University of Bologna              23   
24                  Universidad Complutense de Madrid              21   
25                  Universitat Autònoma de Barcelona              21   
26                                 AOSIS OpenJournals              20   
27                           Universitat de Barcelona              20   
28                                     Ubiquity Press              20   
29                          Universidad de Costa Rica              20   

                                            Publisher  Frequency_2014  
0                      Hindawi Publishing Corporation             438  
1                                      BioMed Central             238  
2                      Scientific Research Publishing             119  
3                                        Bentham open              99  
4                                             MDPI AG              95  
5                                            Springer              95  
6                                  Dove Medical Press              91  
7                                Medknow Publications              80  
8                                  Libertas Academica              49  
9                              PAGEPress Publications              47  
10                                           Versita               46  
11              Internet Scientific Publications, LLC              46  
12                            Copernicus Publications              35  
13                   Universidad Nacional de Colombia              34  
14           Canadian Center of Science and Education              34  
15                                            Versita              31  
16                               Co-Action Publishing              31  
17                         Universidade de Sí£o Paulo              31  
18                                    Frontiers Media              30  
19              Tehran University of Medical Sciences              30  
20                                    Hans Publishers              26  
21                                           GERFLINT              23  
22           Asian Network for Scientific Information              23  
23                                       IACSIT Pres

##Merge
this will try to find out what was the publisher size in 2014. The **Publiser** columns represent the publisher for **2015**

In [45]:
pd.merge(publishers_2015, publishers_2014, how='left', on='Publisher').head(30)

Publisher  Frequency_2015  \
0                      Hindawi Publishing Corporation             539   
1                                      BioMed Central             271   
2                                     De Gruyter Open             212   
3                                             MDPI AG             121   
4                                            Springer             108   
5                                  Dove Medical Press             105   
6                                        Bentham open              81   
7                                Medknow Publications              78   
8                                            Elsevier              72   
9                                  Libertas Academica              56   
10                                    Frontiers Media              49   
11              Internet Scientific Publications, LLC              46   
12                            Copernicus Publications              43   
13                             PAGEPress Publications              42   
14                   Universidad Nacional de Colombia              35   
15                          Universidade de São Paulo              33   
16                                              Wiley              33   
17    Consejo Superior de Investigaciones Científicas              31   
18                               Co-Action Publishing              31   
19                                  SAGE Publications              28   
20              Tehran University of Medical Sciences              28   
21                                           GERFLINT              24   
22  Academy & Industry Research Collaboration Cent...              23   
23                              University of Bologna              23   
24                  Universidad Complutense de Madrid              21   
25                  Universitat Autònoma de Barcelona              21   
26                                 AOSIS OpenJournals              20   
27                           Universitat de Barcelona              20   
28                                     Ubiquity Press              20   
29                          Universidad de Costa Rica              20   

    Frequency_2014  
0              438  
1              238  
2              NaN  
3               95  
4               95  
5               91  
6               99  
7               80  
8                8  
9               49  
10              30  
11              46  
12              35  
13              47  
14              34  
15             NaN  
16              12  
17             NaN  
18              31  
19               9  
20              30  
21              23  
22              19  
23              19  
24              17  
25             NaN  
26              19  
27              17  
28               9  
29              14